packages/dev/core/src/FlowGraph/Blocks/Event/flowGraphMeshPickEventBlock.ts

```diff
@@ -1,38 +1,40 @@
-import type { AbstractMesh } from "../../../Meshes/abstractMesh";
+import { AbstractMesh } from "../../../Meshes/abstractMesh";
 import { FlowGraphEventBlock } from "../../flowGraphEventBlock";
 import { PointerEventTypes } from "../../../Events/pointerEvents";
-import type { FlowGraphContext } from "core/FlowGraph/flowGraphContext";
+import type { FlowGraphContext } from "../../flowGraphContext";
 import type { IFlowGraphBlockConfiguration } from "../../flowGraphBlock";
 import { RegisterClass } from "../../../Misc/typeStore";
+import type { FlowGraphPath } from "../../flowGraphPath";
+import { Tools } from "../../../Misc/tools";
 /**
  * @experimental
  */
 export interface IFlowGraphMeshPickEventBlockConfiguration extends IFlowGraphBlockConfiguration {
-    meshVariableName: string;
+    path: FlowGraphPath;
 }
 /**
  * @experimental
  * A block that activates when a mesh is picked.
  */
 export class FlowGraphMeshPickEventBlock extends FlowGraphEventBlock {
-    private _meshVariableName: string;
-
     public constructor(public config: IFlowGraphMeshPickEventBlockConfiguration) {
+        if (config.path.hasTemplateStrings) {
+            Tools.Warn("Template strings are not supported in the path of mesh pick event blocks.");
+        }

```

Comment:
Is that because this block requires a fixed element to register events on?

Reply:
That's correct